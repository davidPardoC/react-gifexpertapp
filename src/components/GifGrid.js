import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { loading, data:images } = useFetchGifs(category);

  return (
    <>
      <h3 className='animate__animated animate__zoomIn'>{category}</h3>
      {loading ? (
        <p className='animate__animated animate__flash'>loading...</p>
      ) : (
        <div className="card-grid animate__fadeIn">
          {images.map((img) => (
            <GifGridItem key={img.id} {...img} />
          ))}
        </div>
      )}
    </>
  );
};
