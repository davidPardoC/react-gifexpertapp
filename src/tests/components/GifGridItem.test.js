import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("GifGridItem Component", () => {
  const title = "Goku";
  const url = "testUrl";
  let wrapper = shallow(<GifGridItem title={title} url={url} />);

  beforeEach(() => {
    const title = "Goku";
    const url = "testUrl";
    wrapper = shallow(<GifGridItem title={title} url={url} />);
  });

  test("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should check text in paragraph ', () => {
      const paragraphText = wrapper.find('p').text().trim()
      expect(paragraphText).toBe(title)
  })

  test('Img shuld have alt and title ', () => {
      const img = wrapper.find('img')
      expect(img.prop('src')).toBe(url)
      expect(img.prop('alt')).toBe(title)
  })

  test('should have card class', () => {
    const div = wrapper.find('div')
    expect(div.prop('className').includes('card')).toBe(true)
  })
  
  
  
});
