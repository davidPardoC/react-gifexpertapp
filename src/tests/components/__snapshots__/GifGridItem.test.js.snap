// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem Component should match snapshot 1`] = `
<div
  className="card"
>
  <img
    alt="Goku"
    src="testUrl"
  />
  <p>
    Goku
  </p>
</div>
`;
